plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java

```diff
@@ -40,201 +44,52 @@
-     * Adds error mark to host field.
-     */
-    void markHostInvalid();
-
-    /**
-     * Removes error mark from host field.
-     */
-    void unmarkHost();
-
-    /**
-     * Adds error mark to port field.
-     */
-    void markPortInvalid();
+    boolean selectTarget(Target target);
 
     /**
-     * Removes error mark from port field.
+     * Shows hint panel.
      */
-    void unmarkPort();
-
-    /**
-     * Sets SSH user name.
-     *
-     * @param userName
-     *          user name
-     */
-    void setUserName(String userName);
-
-    /**
-     * Returns value of user name field.
-     *
-     * @return
-     *         value of user name field
-     */
-    String getUserName();
-
-    /**
-     * Sets SSH password.
-     *
-     * @param password
-     *          password
-     */
-    void setPassword(String password);
-
-    /**
-     * Returns value of password field.
-     *
-     * @return
-     *          value of password field
-     */
-    String getPassword();
-
-    /**
-     * Enables or disables Save button.
-     *
-     * @param enable
-     *          enabled state
-     */
-    void enableSaveButton(boolean enable);
-
-    /**
-     * Enables or disables Cancel button.
-     *
-     * @param enable
-     *          enabled state
-     */
-    void enableCancelButton(boolean enable);
-
-    /**
-     * Enables or disables Connect button.
-     * @param enable
-     */
-    void enableConnectButton(boolean enable);
+    void showHintPanel();
 
     /**
-     * Changes the text of Connect button.
+     * Sets a properties panel.
      *
-     * @param text
-     *          new text
+     * @param widget
      */
-    void setConnectButtonText(String text);
-
-    /**
-     * Focuses and selects all the text in the Name field.
-     */
-    void selectTargetName();
-
+    void setPropertiesPanel(IsWidget widget);
 
     interface ActionDelegate {
 
-        // Perform actions when clicking Close button
-        void onCloseClicked();
-
         // Perform actions when clicking Add target button
-        void onAddTarget(String category);
+        void onAddTarget(String type);
 
         // Is called when target is deleted
         void onDeleteTarget(Target target);
 
-        // Perform actions when selecting a target
-        void onTargetSelected(Target target);
-
-        void onTargetNameChanged(String value);
-
-        void onHostChanged(String value);
-
-        void onPortChanged(String value);
+        void updateTargets(String preselectTargetName);
 
-        void onUserNameChanged(String value);
+        void selectTarget(Target selectedTarget);
 
-        void onPasswordChanged(String value);
+        // Perform actions when clicking Close button
+        void onCloseClicked();
 
-        void onSaveClicked();
+        // Perform actions when selecting a target
+        void onTargetSelected(Target target);
 
-        void onCancelClicked();
+        boolean isTargetNameExist(String targetName);
```

Comment:
Delegates are used to delegate user events from view to its presenter. And only for that.
Don't use any other methods, only handle events.